app/main.py

```diff
@@ -1,6 +1,39 @@
 from fastapi import FastAPI
 from fastapi.middleware.cors import CORSMiddleware
+from pydantic import BaseModel
+import uvicorn
+from langchain_community.document_loaders import CSVLoader
+from langchain_community.vectorstores import Chroma
+from langchain_community.embeddings import HuggingFaceEmbeddings
+import os
+import traceback
+import requests
+import csv
+import math
 
+EMBEDDING_MODEL = HuggingFaceEmbeddings(model_name="sentence-transformers/distiluse-base-multilingual-cased-v2")
+
+GOOGLE_MAP_GEOCODING_API_KEY = os.getenv("GOOGLE_MAP_GEOCODING_API_KEY")
+
+persist_directory = "./app/vectordb"
+
+csv_filepath = "./app/data/example.csv"
+
+fieldlist = ["id", "name", "long", "lat", "area", "category"]
+
+LONG_INDEX = 2
+LAT_INDEX = 3
+
+async def get_cordinates(place_name: str) -> float:
```

Comment:
このアノテーションでは本番環境時にエラーが走っちゃうかもですね。
返り値が２つある場合はタプル型で返すことが多いです。あと条件によって返り値が異なる場合はUnion型を使用します。
```suggestion
async def get_cordinates(place_name: str) -> Union[Tuple[float, float], Tuple[None, None]]:
```

Reply:
とまあアノテーションが長くなってしまいます。これだとちょっと見栄え悪いですよね？
というわけでこの関数の返り値はモデルで型を定義してみましょう。`app/models/`の何かしらにファイルを作成して`Coordinate`クラスを作成しましょう。そうするとこのようにアノテーションを書き換えることができます。Optionalクラスは[]の型or Noneが返ってくることを意味します
```suggestion
async def get_cordinates(place_name: str) -> Optional[Coordinate]:
```

Reply:
ついでにですが関数名も見直した方がいいかもですね。この関数は広義関数ではなく実用関数なので具体で名前をつけます。あとスペルも気をつけてみてください。読みやすさが桁違いです。
**※最終的にディレクトリ分けしたときにはまた広義関数に戻す(抽象クラスにぶち込む)のでこのままでもいいっちゃいいです。好みの問題ってやつです←**
```suggestion
async def get_izakaya_coordinate(place_name: str) -> Optional[Coordinate]:
```

Reply:
> というわけでこの関数の返り値はモデルで型を定義してみましょう。

Coodinateクラスを`app/models/coodinate.py`に下の感じで定義した！
```
from pydantic import BaseModel
class Coordinate(BaseModel):
    lat: float
    lng: float
```

アノテーションはこんな感じでつけた！！
この記事で見た感じはこっちの方がわかりやすそうやなって思った！
https://qiita.com/Yossy_Hal/items/2feee8d6c91f9270f266

```
async def get_izakaya_coordinates(place_name: str) -> Coordinate | None:
```

Reply:
この書き方知らなかったですが結構いいですねー
補足ですが、OptionalやUnionはモデル自体に複数の型を用意できるので便利ですね
例えば`Coordinate`自体を
```py
class Coordinate(BaseModel):
    lat: Optional[float] = None
    lat: Union[float, None]
    lng: Optional[float] = None
    lng: Optiona[float, None]
```
みたいな表し方（スキーマは適当です）をしておいて
```py
async def get_search_point_coordinates(place_name: str) -> Coordinate:
```
だけでNone表現ができたりします〜